contracts/NFTVesting/IntervalVestingNFTTimeLock.sol

```diff
@@ -0,0 +1,169 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts (last updated v4.5.0) (token/ERC20/utils/TokenTimelock.sol)
+// SpartanLabs Contracts (NFTVesting)
+
+pragma solidity ^0.8.0;
+import "./IERC721.sol";
+
+/**
+ * @dev A single NFT holder contract that will allow a beneficiary to extract the
+ * NFT after a given release time.
+ *
+ * Useful for simple vesting schedules like "whitelisted addresses get their NFT
+ * after 1 year".
+ */
+contract IntervalVestingNFTTimeLock {
+    // ERC721 basic token smart contract
+    IERC721 private immutable _nft;
+
+    // ERC721 basic token ID of contract being held
+    uint256 private immutable _tokenId;
+
+    // beneficiary of token after they are released
+    address private immutable _beneficiary;
+
+    // timestamp when token release is enabled
+    uint256 private immutable _releaseTime;
+
+    // Max discount allowed for a token in percentage
+    uint256 private immutable _maxDiscount;
+
+    // Max number of Interval for vesting
+    uint256 private immutable _maxInterval;
+
+    // Current interval after @variable _releaseTime
+    uint256 private _currentInterval;
+
+    // Duration for each Interval
+    uint256 private _intervalDuration;
+
+    modifier validRelease() {
+        require(
+            block.timestamp >= _releaseTime + _intervalDuration,
+            "Vesting Schedule is not Up yet."
+        );
+        _;
+    }
+
+    /**
+     * @dev Deploys a timelock instance that is able to hold the token specified, and will only release it to
+     * `beneficiary_` when {release} is invoked after `releaseTime_`. The release time is specified as a Unix timestamp
+     * (in seconds).
+     */
+    constructor(
+        IERC721 nft_,
+        uint256 tokenId_,
+        address beneficiary_,
+        uint256 releaseTime_,
+        uint256 maxDiscount_,
+        uint256 maxInterval_,
+        uint256 intervalDuration_
+    ) {
```

Comment:
Same - Perhaps this should also have a requirement that when constructing that it has been given enough ETH to constitute the discount?

Reply:
okay, necessary check

Reply:
added check